src/Std/Data/Iterators/Basic.lean

```diff
@@ -203,6 +228,37 @@ def PlausibleIterStep.done {IsPlausibleStep : IterStep α β → Prop}
     (h : IsPlausibleStep .done) : PlausibleIterStep IsPlausibleStep :=
   ⟨.done, h⟩
 
+/--
+An inductive type that makes it easier to apply the `cases` tactic to a
+`PlausibleIterStep`. See `PlausibleIterStep.casesHelper` for more information.
+-/
+inductive PlausibleIterStep.CasesHelper {IsPlausibleStep : IterStep α β → Prop} :
+    PlausibleIterStep IsPlausibleStep → Type _ where
+  | yield (it' out h) : CasesHelper ⟨.yield it' out, h⟩
+  | skip (it' h) : CasesHelper ⟨.skip it', h⟩
+  | done (h) : CasesHelper ⟨.done, h⟩
+
+/--
+Because `PlausibleIterStep` is a subtype of `IterStep`, it is tedious to use
+the `cases` tactic:
+
+```lean
+obtain ⟨step, h⟩ := step
+cases step
+```
+
+Using `casesHelper`, the case distinction can be done more ergonomically.
+
+```lean
+cases step.casesHelper
+```
+-/
+def PlausibleIterStep.casesHelper {IsPlausibleStep : IterStep α β → Prop} :
```

Comment:
I think the intended way to do this is to define a custom `casesOn` for `PlausibleIterStep` and tag it with `@[cases_eliminator]`. Then `cases step` will do the right thing and you don't need `casesHelper`.